tee-worker/omni-executor/native-task-handler/src/native_call_handlers.rs

```diff
@@ -91,7 +91,24 @@ pub async fn handle_native_call<
 				.omni_account()
 				.auth_token_requested(AccountId32(omni_account.into()), claims.exp);
 
-			let tx = ctx.transaction_signer.sign(auth_token_requested_call, None).await;
+			// Without increase nonce, all requests after request_auth_token will failure with below error.
+			// Could not submit tx: Rpc(ClientError(Call(ErrorObject { code: ServerError(1014), message: "Priority is too low: (2564 vs 2564)",
+			// data: Some(RawValue("The transaction has too low priority to replace another transaction already in the pool.")) })))
+			let signer_account_id = ctx.transaction_signer.get_signer_account_id();
+			let nonce = match rpc_client.get_account_nonce(&signer_account_id).await {
+				Ok(n) => n,
+				Err(e) => {
+					log::error!("Failed to get account nonce: {:?}", e);
+					let response =
+						NativeOperationResponse::Err(NativeOperationError::InternalError);
+					if response_sender.send(response.encode()).is_err() {
+						log::error!("Failed to send response");
+					}
+					return;
+				},
+			};
+			// Increment nonce for the next transaction
+			let tx = ctx.transaction_signer.sign(auth_token_requested_call, Some(nonce + 1)).await;
```

Comment:
If I look up `ctx.transaction_signer.sign` in the file, I see different code:
- `ctx.transaction_signer.sign(.., Some(nonce) ...)`
- `ctx.transaction_signer.sign(.., Some(nonce + 1) ...)`
- `ctx.transaction_signer.sign(.., None ...)`

Which one is correct? Can we have consistent code for it?
@wli-pro @silva-fj

Reply:
We should use `None` when we submit only one extrinsic, the signer will get the current nonce and set it. But in cases where we submit more than once (e.g: `RequestIntent`), we need to increment the nonce manually for the second extrinsic. This is where we use the `Some()` variant. (In the first two examples we are doing the same thing, it's just that in the first one, the nonce is incremented before passing it to the signer and the second one it is incremented while passing it to the signer.

Another option could be to have two methods. `sign(..)` and `sign_with_nonce(.., nonce)`, but the logic should be the same

Reply:
This is something that will very likely cause troubles when more than one thread is using same signer account.

Reply:
True, we should probably track the nonce in-memory

Reply:
I'll create an issue for this